plugins/woocommerce/includes/class-wc-discounts.php

```diff
@@ -257,7 +256,7 @@ public function apply_coupon( $coupon, $validate = true ) {
 		}
 
 		$coupon_code = $coupon->get_code();
-		if ( StringUtil::is_null_or_whitespace( $this->discounts[ $coupon_code ] ?? null ) ) {
+		if ( ! isset( $this->discounts[ $coupon_code ] ) || ! is_array( $this->discounts[ $coupon_code ] ) ) {
```

Comment:
We talked a bit about this a bit and figured out that the `is_null_or_whitespace` was probably added here in error while fixing a different issue. We want to ensure here that `$this->discounts[ $coupon_code ]` is not yet a valid array, so that we can make it one without overwriting anything in the next line.